src/frontend/src/lib/components/loaders/LoaderUserProfile.svelte

```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { authIdentity } from '$lib/derived/auth.derived';
+	import { loadUserProfile } from '$lib/services/load-user-profile.services';
+	import type { OptionIdentity } from '$lib/types/identity';
+
+	let identity: OptionIdentity;
+	$: identity = $authIdentity;
+
+	onMount(() => {
+		loadUserProfile({ identity });
```

Comment:
It works because there are guards but, are we sure at this point that the autosubscriber really kicked in?

```
let identity: OptionIdentity;
$: identity = $authIdentity; // <--- Did this happen?
	
onMount(() => {
  loadUserProfile({ identity }); // <- Was is really set
```

To be on the safe side, I would listen to the auto subscription, reset the store if identity is undefined else load.

WDYT?

Reply:
I don't know if I understood correctly, have a look at latest commit, is that what you had in mind?

Reply:
Almost, following is missing:

```
const load = async ({ reload = false }: { reload?: boolean }) => {

                // In this function I would just use the same store as the one subscribed, just in case of race condition
                // I doubt it happens, just because I'm paranoiac

		if (isNullish($authIdentity)) {
			userProfileStore.reset();
			return;
		}
		
		loadUserProfile({ identity: $authIdentity, reload });
	};
	
// Remove onMount

// Auto subscribe to trigger load
$: $authIdentity, load()
```

Reply:
Ah! understood! A guard on the `authIdentity`!

Reply:
done, have a look please